src/hooks/browser-headers.ts

```diff
@@ -1,26 +1,52 @@
-const http2 = require('http2-wrapper');
+import { URL } from 'url';
+import { Options } from 'got-cjs';
+import http2 = require('http2-wrapper');
+import { Context } from '../context';
 
 /**
- * @param {object} options
+ * Merges original generated headers and user provided overrides.
+ * All header overrides will have the original header case, because of antiscraping.
  */
-exports.browserHeadersHook = async function (options) {
+export function mergeHeaders(original: Record<string, string>, overrides: Record<string, string>) {
+    const fixedHeaders = new Map();
+
+    for (const entry of Object.entries(original)) {
+        fixedHeaders.set(entry[0].toLowerCase(), entry);
+    }
+
+    for (const entry of Object.entries(overrides)) {
+        fixedHeaders.set(entry[0].toLowerCase(), entry);
+    }
+
+    const headers: Record<string, string> = {};
+    for (const [key, value] of fixedHeaders.values()) {
+        headers[key] = value;
+    }
+
+    return headers;
```

Comment:
Can be converted into this:
```suggestion
    return Object.fromEntries(fixedHeaders.values());
```

Reply:
Thanks, didn't know about this :D